### PR TITLE
Smoke Test Updates (Remove Sprout Logic )

### PR DESCRIPTION
**Pull request by mdr0id**
Closes and cleans up https://github.com/zcash/zcash/pull/5239

### CODE REVIEW

**Review by daira — APPROVED** [review]
utACK

File: qa/zcash/smoke_tests.py [hunk 1]
```diff
@@ -433,193 +403,75 @@ def transaction_chain(zcash):
...
-        # multiple Sprout -> taddr
         # multiple Sapling -> taddr
-        check_z_mergetoaddress_parallel(None, zcash, [
-            ('', [sprout_zaddr_1, sprout_zaddr_2], taddr_1, sprout_balance - DEFAULT_FEE),
-            ('', [sapling_zaddr_1, sapling_zaddr_2], taddr_2, sapling_balance - DEFAULT_FEE),
-        ])
-        taddr_balance = sprout_balance + sapling_balance - (Decimal('2') * DEFAULT_FEE)
-        sprout_balance = Decimal('0')
+        check_z_mergetoaddress(None, '', zcash, [sapling_zaddr_1, sapling_zaddr_2], taddr_2, sapling_balance - DEFAULT_FEE)
+        taddr_balance = sapling_balance - (Decimal('2') * DEFAULT_FEE)
```
Comment (str4d) [inline qa/zcash/smoke_tests.py:469]:
```suggestion
        taddr_balance += sapling_balance - (Decimal('2') * DEFAULT_FEE)
```

File: qa/zcash/smoke_tests.py [hunk 2]
```diff
@@ -433,193 +403,75 @@ def transaction_chain(zcash):
...
-        taddr_1_balance = Decimal(zcash.z_getbalance(taddr_1)).quantize(Decimal('1.00000000'))
         taddr_2_balance = Decimal(zcash.z_getbalance(taddr_2)).quantize(Decimal('1.00000000'))
-        check_z_mergetoaddress_parallel(results, zcash, [
-            ('4y', [taddr_1], sprout_zaddr_1, taddr_1_balance - DEFAULT_FEE),
-            ('4cc', [taddr_2], sapling_zaddr_1, taddr_2_balance - DEFAULT_FEE),
-        ])
-        sprout_balance = taddr_1_balance - DEFAULT_FEE
+        check_z_mergetoaddress(results, '4cc', zcash,  [taddr_2], sapling_zaddr_1, taddr_2_balance - DEFAULT_FEE)
         sapling_balance = taddr_2_balance - DEFAULT_FEE
         taddr_balance = Decimal('0')
```
Comment (str4d) [inline qa/zcash/smoke_tests.py:476]:
```suggestion
        taddr_balance -= taddr_2_balance
```

**Review by str4d — CHANGES_REQUESTED** [review]

**Comment by mdr0id** [comment]
@str4d Updated comments and added brief usage doc

**Review by daira — DISMISSED** [review]
re-utACK b6987844f8ed7d39b3f09d6b4a913f9ce76bda0c

File: qa/zcash/smoke_tests.py [hunk 3]
```diff
@@ -433,195 +409,75 @@ def transaction_chain(zcash):
...
+        check_z_sendmany(results, '4v', zcash, sapling_zaddr_2, [
                 (taddr_4, (starting_balance / Decimal('10'))),
-                (taddr_5, (starting_balance / Decimal('10'))),
-            ]),
-        ])
-        sprout_balance -= ((starting_balance / Decimal('10')) * Decimal('2')) + DEFAULT_FEE
+                (taddr_5, (starting_balance / Decimal('10')))])[0]
+            
         sapling_balance -= ((starting_balance / Decimal('10')) * Decimal('2')) + DEFAULT_FEE
         taddr_balance += (starting_balance / Decimal('10')) * Decimal('4')
```
Comment (str4d) [inline qa/zcash/smoke_tests.py:450]:
```suggestion
        taddr_balance += (starting_balance / Decimal('10')) * Decimal('2')
```

File: qa/zcash/smoke_tests.py [hunk 4]
```diff
@@ -433,195 +409,75 @@ def transaction_chain(zcash):
...
-        # multiple taddr -> Sprout
         # multiple taddr -> Sapling
-        check_z_mergetoaddress_parallel(results, zcash, [
-            ('4z', [taddr_1, taddr_3], sprout_zaddr_2, sprout_balance + ((starting_balance / Decimal('10')) * Decimal('2')) - DEFAULT_FEE),
-            ('4bb', [taddr_4, taddr_5], sapling_zaddr_2, sapling_balance + ((starting_balance / Decimal('10')) * Decimal('2')) - DEFAULT_FEE),
-        ])
-        sprout_balance += ((starting_balance / Decimal('10')) * Decimal('2')) - DEFAULT_FEE
+        check_z_mergetoaddress(results, '4bb',zcash, [taddr_4, taddr_5], sapling_zaddr_2, sapling_balance + ((starting_balance / Decimal('10')) * Decimal('2')) - DEFAULT_FEE)
         sapling_balance += ((starting_balance / Decimal('10')) * Decimal('2')) - DEFAULT_FEE
         taddr_balance -= (starting_balance / Decimal('10')) * Decimal('4')
```
Comment (str4d) [inline qa/zcash/smoke_tests.py:455]:
```suggestion
        taddr_balance -= (starting_balance / Decimal('10')) * Decimal('2')
```

File: qa/zcash/smoke_tests.py [hunk 5]
```diff
@@ -433,195 +409,75 @@ def transaction_chain(zcash):
...
-        # multiple Sprout -> taddr
         # multiple Sapling -> taddr
-        check_z_mergetoaddress_parallel(None, zcash, [
-            ('', [sprout_zaddr_1, sprout_zaddr_2, sprout_zaddr_3], taddr_1, sprout_balance - DEFAULT_FEE),
-            ('', [sapling_zaddr_1, sapling_zaddr_2, sapling_zaddr_3], taddr_2, sapling_balance - DEFAULT_FEE),
-        ])
-        taddr_balance = sprout_balance + sapling_balance - (2 * DEFAULT_FEE)
-        sprout_balance = Decimal('0')
+        check_z_mergetoaddress(None, '', zcash, [sapling_zaddr_1, sapling_zaddr_2, sapling_zaddr_3], taddr_2, sapling_balance - DEFAULT_FEE)
+        taddr_balance += sapling_balance - (2 * DEFAULT_FEE)
```
Comment (str4d) [inline qa/zcash/smoke_tests.py:459]:
```suggestion
        taddr_balance += sapling_balance - DEFAULT_FEE
```

File: qa/zcash/smoke_tests.py [hunk 6]
```diff
@@ -433,195 +409,75 @@ def transaction_chain(zcash):
...
-        # multiple Sprout -> taddr
         # multiple Sapling -> taddr
-        check_z_mergetoaddress_parallel(None, zcash, [
-            ('', [sprout_zaddr_1, sprout_zaddr_2], taddr_1, sprout_balance - DEFAULT_FEE),
-            ('', [sapling_zaddr_1, sapling_zaddr_2], taddr_2, sapling_balance - DEFAULT_FEE),
-        ])
-        taddr_balance = sprout_balance + sapling_balance - (Decimal('2') * DEFAULT_FEE)
-        sprout_balance = Decimal('0')
+        check_z_mergetoaddress(None, '', zcash, [sapling_zaddr_1, sapling_zaddr_2], taddr_2, sapling_balance - DEFAULT_FEE)
+        taddr_balance += sapling_balance - (Decimal('2') * DEFAULT_FEE)
```
Comment (str4d) [inline qa/zcash/smoke_tests.py:473]:
```suggestion
        taddr_balance += sapling_balance - DEFAULT_FEE
```

**Review by str4d — CHANGES_REQUESTED** [review]
I double-checked the math around the PR changes.

**Review by str4d — APPROVED** [review]
utACK

**Pull request merged**